README.md

```diff
@@ -40,10 +38,63 @@ This project is a full-stack web application built with Django and React. It con
 - Backend API: http://localhost:8000
 - Frontend UI: http://localhost:3000
 
-## Prerequisites
+### Interacting with docker
+Since docker is a containerized environment, to interact with docker you need to run commands inside the given docker container.
+
+This can be done like this:
+```sh
+docker compose exec <Name of container> <command to run>
+```
+
+To find the name of the container look into `docker-compose.yml`. The names of the containers at the time of writing is:
+- db
+- backend
+- frontend
+
+Command to run can be whatever. For instance to drop into a shell in the container backend run:
+```sh
+docker compose exec backend sh
+```
+### Prerequisites
 - Docker
 - Docker Compose   
 
+## Pushing changes to the cloud
+Be sure to be in the root of the project. This is the directory which contains the `docker-compose.yml` file. 
+
+### Before starting - Do this once
+Before starting it is necesarry to make sure that you have the correct programs installed.
+#### Install google cloud CLI tools
+Follow [this link](https://cloud.google.com/sdk/docs/install) and follow the instructions to install the google cloud CLI on your system.
+
+#### Install docker
+Follow [this link](https://docs.docker.com/engine/install/) and follow the instructions to install docker for your system.
+
+## How to push the changes to google cloud
+To make this process simpler, a script named `push_to_cloud.sh` has been created. After cloning the project this script might not be runnable. To make it runnable run this command:
+```sh
+sudo chmod +x push_to_cloud.sh
+```
+
+After doing this you should be able to run the script:
+```sh
+./push_to_cloud.sh
+```
+
+This script will first ask you to login to google by opening a browser window. Login to the user you have which is connected to the `dogplus_polimi` project on google cloud. After doing this the script builds the docker images for the backend and pushes this to the cloud. It then does the same for the frontend
+
+## IMPORTANT:
+After running the script successfully, the frontend part of the application is not yet correctly published. To finish the process, follow these steps:
+1. Go to [the cloud run dashboard](https://console.cloud.google.com/run?project=dog-plus-polimi)
+2. Click 'Edit and deply new reviosion'.
+3. Edit the 'Container Image URL' with the container you just created. This should be under 'Artifact Registry' -> 'europe-west8-docker.pkg.dev/dog-plus-polimi/dogplus-frontend-repository' -> 'dogplus-react-app'
```

Comment:
Where exactly is the artifact registry

Reply:
Have added screenshots to make this clearer